### PR TITLE
Hmis 174

### PR DESCRIPTION
**Pull request by subrotomukherjee**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/HMIS-174

### Change description ###
Adding unit tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/resources/application-local.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+targetInstance: https://hmi-apim-svc-test.azure-api.net
+targetHost:  hmi-apim-svc-test.azure-api.net
+targetSubscriptionKey: 587c56f59a604054b1ce73dac0ea5b28
```
Comment (karanotts) [inline src/test/resources/application-local.yaml:3]: Please, no secrets in commits!

File: azure-pipelines.yml [hunk 2]
```diff
@@ -128,7 +128,26 @@ stages:
...
+    jobs:
+    - job:
+      pool:
+        vmImage: 'windows-latest'
+      steps:
+      - task: Gradle@2
+        inputs:
+          gradleWrapperFile: 'gradlew'
+          workingDirectory: '$(System.DefaultWorkingDirectory)'
+          tasks: clean test -DTEST_SUBSCRIPTION_KEY=$(hmi-apim-svc-sbox-sub-key) -DTEST_URL=https://hmi-apim-svc-sbox.azure-api.net -DTEST_HOST=hmi-apim-svc-sbox.azure-api.net
```
Comment (karanotts) [inline azure-pipelines.yml:143]: If this is the only bit different from functional and smoke, please modularise as template, doesn't look good to copy tens of lines of code for just one line.

Comment (karanotts) [reply azure-pipelines.yml:143]: Also, can we stick to convention "clean functional" "clean smoke" and rename this to match, like "clean unit" to indicate what kind of test is running?

Comment (subrotomukherjee) [reply azure-pipelines.yml:115]: I take your first point.
On second point I however I kind of disagree, as by convention the unit tests for a project reside in the test directory and the gradle task is named accordingly. 
So I would like to keep it as it is.

File: src/test/resources/application-local.yaml (hunk 1 above)
Comment (subrotomukherjee) [reply src/test/resources/application-local.yaml]: Removed now.

File: src/test/resources/application-local.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+targetInstance: https://hmi-apim-svc-test.azure-api.net
+targetHost:  hmi-apim-svc-test.azure-api.net
+targetSubscriptionKey: samplesecret
+hearingApiRootContext: /hmi/hearings
+hmiApiRootContext : /hmi/
+spring.flyway.enabled: false
```
Comment (santhoshhmcts) [inline src/test/resources/application-local.yaml:6]: do you need this, we don't hold any schemas to migrate right?

File: src/test/resources/application-test.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+targetInstance: ${TEST_URL}
+targetHost: ${TEST_HOST}
+targetSubscriptionKey: ${TEST_SUBSCRIPTION_KEY}
+hearingApiRootContext: /hmi/hearings
+hmiApiRootContext : /hmi/
+spring.flyway.enabled: false
```
Comment (santhoshhmcts) [inline src/test/resources/application-test.yaml:6]: same as above comment

File: src/test/resources/requests/allocated-listing-team-missing-request.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (santhoshhmcts) [inline src/test/resources/requests/allocated-listing-team-missing-request.json:1]: Use a single (templated) version of these request json files and dynamically modify the values as your stubs, this way we have less code to maintain.

File: src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/util/ResponseVerifier.java [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+                            "    \"Name\": \"HMCTS\",\n" +
+                            "    \"Place\": \"London\"\n" +
+                            "}",
+                    response.getBody().asString(), JSONCompareMode.STRICT);
+        } catch (JSONException jsonException) {
+            throw new AssertionError("Payloads have not matched");
+        }
+
+
+        //Option 3 - Use a better Third Party Specialised Matcher (Hamcrest)
```
Comment (santhoshhmcts) [inline src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/util/ResponseVerifier.java:40]: remove this comment

File: build.gradle [hunk 7]
```diff
@@ -277,12 +284,29 @@ dependencies {
...
-          "org.junit.vintage:junit-vintage-engine:5.6.2",
-          "org.assertj:assertj-core:${assertJVersion}"
+  testImplementation group: 'org.springframework.boot', name: 'spring-boot-starter', {
+    exclude group: 'org.hamcrest', module: 'hamcrest-core'
+  }
+
+  testCompile group: 'org.springframework.boot', name: 'spring-boot-starter', {
+    exclude group: 'org.hamcrest', module: 'hamcrest-core'
+  }
+  testCompile ("org.springframework.boot:spring-boot-starter-test")
```
Comment (santhoshhmcts) [inline build.gradle:294]: are we using all these libraries? if not remove unused ones

Comment (subrotomukherjee) [reply build.gradle:294]: Yes

File: src/test/resources/application-local.yaml (hunk 3 above)
Comment (subrotomukherjee) [reply src/test/resources/application-local.yaml:6]: Yes. It makes local development easy. It's a standard practice with spring boot.

File: src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/util/ResponseVerifier.java (hunk 6 above)
Comment (subrotomukherjee) [reply src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/util/ResponseVerifier.java:40]: Done.

File: src/test/resources/application-test.yaml (hunk 4 above)
Comment (subrotomukherjee) [reply src/test/resources/application-test.yaml]: Removed

File: src/test/resources/application-local.yaml (hunk 3 above)
Comment (subrotomukherjee) [reply src/test/resources/application-local.yaml]: Removed.

File: src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/steps/HearingApiCallSteps.java [hunk 8]
```diff
@@ -1,16 +1,15 @@
...
+import static io.restassured.RestAssured.expect;
+
 import io.restassured.response.Response;
-import net.thucydides.core.annotations.Step;
 
 import java.util.Map;
 
-import static net.serenitybdd.rest.SerenityRest.expect;
 
 public class HearingApiCallSteps {
```
Comment (santhoshhmcts) [inline src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/steps/HearingApiCallSteps.java:10]: I dont think we need this class now, pull this method up and remove this class if not required.

File: src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/steps/HearingApiCallSteps.java [hunk 9]
```diff
@@ -1,16 +1,15 @@
...
     private String actor;
 
-    @Step("#actor routes to {0} in order to get invoke {1}")
     public Response requestHearingWithMissingField(final String api, final Map<String, Object> headersAsMap, final String basePath, final String payloadBody) {
```
Comment (santhoshhmcts) [inline src/test/java/uk/gov/hmcts/futurehearings/hmi/unit/testing/steps/HearingApiCallSteps.java:13]: same as above

**Pull request merged**